src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -108,40 +109,55 @@ private static void EncodeToUtf16_Ssse3(ReadOnlySpan<byte> bytes, Span<char> cha
                                  (byte)'8', (byte)'9', (byte)'a', (byte)'b',
                                  (byte)'c', (byte)'d', (byte)'e', (byte)'f');
 
+
+            nuint lengthSubVector128 = (nuint)bytes.Length - (nuint)Vector128<int>.Count;
```

Comment:
This probably deserves a comment summing that its used to ensure we can process the trailing elements as vectorized and that we won't double compute for any evenly divisible by 4 length since we compare `pos > lengthSubVector128` rather than `pos >= lengthSubVector128`